cdap-common/src/main/java/co/cask/cdap/common/ServiceException.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common;
+
+import co.cask.cdap.api.common.HttpErrorStatusProvider;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ * Similar to Exception, but responds with the exception's message, instead of the root cause's message.
+ */
+public class ServiceException extends Exception implements HttpErrorStatusProvider {
+
+  public ServiceException(Throwable cause) {
+    super(cause);
+  }
+
+  @Override
+  public int getStatusCode() {
+    return HttpResponseStatus.INTERNAL_SERVER_ERROR.getCode();
+  }
+
+  @Override
+  public String getMessage() {
+    return getMessage();
```

Comment:
This is a infinite recursion? I don't think you need to override this method.

Reply:
right; removed the overriding implementation.